backend/src/test/java/com/redhat/cloud/notifications/processors/email/EmailTest.java

```diff
@@ -71,17 +72,22 @@
     @RestClient
     RbacServiceToService rbacServiceToService;
 
+    // InjectSpy allows us to update the fields via reflection (Inject does not)
+    @InjectSpy
+    EmailSender emailSender;
+
     static final String BOP_TOKEN = "test-token";
     static final String BOP_ENV = "unitTest";
     static final String BOP_CLIENT_ID = "test-client-id";
 
     @BeforeAll
     void init() {
         String url = String.format("http://%s/v1/sendEmails", mockServerConfig.getRunningAddress());
-        System.setProperty("processor.email.bop_url", url);
-        System.setProperty("processor.email.bop_apitoken", BOP_TOKEN);
-        System.setProperty("processor.email.bop_env", BOP_ENV);
-        System.setProperty("processor.email.bop_client_id", BOP_CLIENT_ID);
+
+        TestHelpers.updateField(emailSender, "bopUrl", url, EmailSender.class);
+        TestHelpers.updateField(emailSender, "bopApiToken", BOP_TOKEN, EmailSender.class);
+        TestHelpers.updateField(emailSender, "bopEnv", BOP_ENV, EmailSender.class);
+        TestHelpers.updateField(emailSender, "bopClientId", BOP_CLIENT_ID, EmailSender.class);
```

Comment:
Yes, setting the property is a system-wide setting (it affects all the tests), so I added the reflection calls to keep them locally.

Reply:
Clearing the properties with `System.clearProperty(...)` at the end of the tests didn't work?

Reply:
Just tried again, they don't seem to work well together (I have other test that touches the bopUrl). 

IIUC, the reason is that quarkus starts only once, and once a config is used, the value won't change.
From quarkus test profiles [1]:

> The first time one of these tests is run the Quarkus test extension will start Quarkus. Quarkus will then remain running for the duration of the test run. This makes testing very fast, because Quarkus is only started once, however it does limit you to testing a single configuration per module, as you can’t restart Quarkus in a different configuration. 


Other way to do it would be to use test configuration - although I'm not sure it would help as we need some dynamic values and maybe it would make the tests slower.

[1] https://quarkus.io/blog/quarkus-test-profiles/

Reply:
Thanks for checking! Reflection is fine then 😃